src/npc/npc_update.cpp

```diff
@@ -1756,7 +1756,7 @@ void UpdateNPCs()
                                 }
                                 else if(NPCIsCheep[NPC[A].Type] && NPC[A].Special == 1 && NPC[A].Special5 == 1)
                                     NPC[A].Location.SpeedY += Physics.NPCGravity * 0.6;
-                                else if(NPC[A].Type == 278) // Here was a junk: duplicated "|| NPC[A].Type == 278" condition
+                                else if(NPC[A].Type == 278 || NPC[A].Type == 279)
```

Comment:
You fixed the vanilla bug. That means you also need to make the boolean field at the compat.ini (see the compat.cpp and the compat.h files) to allow disabling this fix. Every fix of vanilla bug must be disablable by compat.ini or compatibility modes.

Reply:
I.e. After making a flag, you can make the next condition:
```
else if(NPC[A].Type == 278 || (g_compatibility.fix_flamethrower_gravity && NPC[A].Type == 279))
```

Note, the boolean `fix_flamethrower_gravity` should be defined at the compat.h file.